src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import getPreferredCurrency from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+
+    /* Onyx Props */
+    // Url currently in view
+    currentURL: PropTypes.string,
+
+    // IOU modal data
+    iouData: PropTypes.shape({
+        isLoadingCurrency: PropTypes.bool.isRequired,
+        isLoadingParticipants: PropTypes.bool.isRequired,
+        createReportInProgress: PropTypes.bool.isRequired,
+    }),
+};
+
+const StepType = {
+    IOUAmount: 'IOUAmount',
+    IOUParticipants: 'IOUParticipants',
+    IOUConfirm: 'IOUConfirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+    iouData: {
```

Comment:
Also don't think this is a great prop name... but I can't think of anything better.

Reply:
This is no longer relevant with the latest changes...

But this was attempting to follow our existing pattern of [grouping props retrieved from Onyx as objects](https://github.com/Expensify/Expensify.cash/blob/7051c27056355f802400080e794a7330ad9191b7/src/pages/home/sidebar/SidebarLinks.js#L33-L58), rather than breaking out each key individually.